siriuspy/siriuspy/devices/fofb_acq.py

```diff
@@ -0,0 +1,1314 @@
+"""FOFB acquisition devices."""
+
+import time as _time
+import numpy as _np
+
+from ..search import PSSearch as _PSSearch, BPMSearch as _BPMSearch
+from ..csdev import Const as _Const
+from ..fofb.csdev import NR_BPM
+
+from .device import ProptyDevice as _ProptyDevice, Devices as _Devices, \
+    Device as _Device
+from .fofb import FOFBCtrlBase as _FOFBCtrlBase
+from .timing import Event
+from .psconv import StrengthConv
+
+
+class _FOFBCtrlAcqConst(_Const):
+    """FOFB acquisition constants."""
+
+    CHANNEL = ('lamp', 'invalid', 'sysid', 'sysid_applied')
+    TRIGTYP = ('now', 'external', 'data', 'software')
+    TRIGEVT = ('start', 'stop')
+    REPEAT = ('normal', 'repetitive')
+    STATES = (
+        'Idle', 'Waiting', 'External Trig', 'Data Trig', 'Software Trig',
+        'Acquiring', 'Error', 'Bad Post Samples', 'Too Many Samples',
+        'No Samples', 'No Memory', 'Overflow')
+
+    Channel = _Const.register('Channel', CHANNEL)
+    TrigTyp = _Const.register('TrigTyp', TRIGTYP)
+    TrigEvt = _Const.register('Events', TRIGEVT)
+    Repeat = _Const.register('Repeat', REPEAT)
+    States = _Const.register('States', STATES)
```

Comment:
Would it make sense to add a TODO item to remind us to turn this into a superclass to be used by BPMs once we update them?

Reply:
agreed, done!